File: Solution/1779.找到最近的有相同X或Y坐标的点/1779.找到最近的有相同X或Y坐标的点.java
/*
 * @Descroption: LeetCode 1779. 找到最近的有相同 X 或 Y 坐标的点
 * @Author: EmoryHuang
 * @Date: 2022-12-01 09:39:35
 * @Method:
 * 遍历坐标数组，在有效坐标中找到曼哈顿距离最小的位置
 */

class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int ans = -1, min_d = 0xffffff;
        for (int i = 0; i < points.length; i++) {
            if (x == points[i][0] || y == points[i][1]) {
                int d = Math.abs(x - points[i][0]) + Math.abs(y - points[i][1]);
                if (d < min_d) {
                    ans = i;
                    min_d = d;
                }
            }
        }
        return ans;
    }
}

File: Solution/1779.找到最近的有相同X或Y坐标的点/1779.找到最近的有相同X或Y坐标的点.py
'''
Descroption: LeetCode 1779. 找到最近的有相同 X 或 Y 坐标的点
Author: EmoryHuang
Date: 2022-12-01 09:30:02
Method:
遍历坐标数组，在有效坐标中找到曼哈顿距离最小的位置
'''


class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        ans = -1
        min_d = 0xffffff
        for i, (a, b) in enumerate(points):
            if x == a or y == b:
                d = abs(x - a) + abs(y - b)
                if d < min_d:
                    ans = i
                    min_d = d
        return ans
